src/main/java/io/thinkit/edc/client/connector/services/ManagementApiHttpClient.java

```diff
@@ -0,0 +1,147 @@
+package io.thinkit.edc.client.connector.services;
+
+import static io.thinkit.edc.client.connector.utils.Constants.ID;
+import static io.thinkit.edc.client.connector.utils.JsonLdUtil.*;
+
+import com.apicatalog.jsonld.JsonLdError;
+import com.apicatalog.jsonld.document.JsonDocument;
+import io.thinkit.edc.client.connector.model.ApiErrorDetail;
+import io.thinkit.edc.client.connector.model.Result;
+import jakarta.json.JsonArray;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+
+public class ManagementApiHttpClient {
+    private final String url;
+    private final HttpClient httpClient;
+    private final UnaryOperator<HttpRequest.Builder> interceptor;
+
+    public ManagementApiHttpClient(String url, HttpClient httpClient, UnaryOperator<HttpRequest.Builder> interceptor) {
+        this.url = url;
+        this.httpClient = httpClient;
+        this.interceptor = interceptor;
+    }
+
+    public String getUrl() {
```

Comment:
I suggested to pass the URL to the `ManagementApiHttpClient`, but it looks like there's no possibility to separate the base url from the path in the `HttpRequest.Builder`, so at this point I will maybe let the base url as a controller field and not pass it here (because it's not used internally).
The alternative would be to have the `send` and `sendAsync` accept a `Function<String, HttpRequest.Builder>` and the string will be the base URL, it will make the signature more complex but at that point the URL will be completely incapsulated.